src/modules/hmis/components/EnrollmentStatus.tsx

```diff
@@ -2,75 +2,88 @@ import ErrorOutlineIcon from '@mui/icons-material/ErrorOutline';
-  | 'error'
-  | 'activeStatus'
-  | 'text.secondary'
-  | 'text.primary';
 
-const EnrollmentStatus = ({
-  enrollment,
-  hideIcon = false,
-  withActiveRange = false,
-  activeColor = 'activeStatus',
-  closedColor = 'text.secondary',
-}: {
-  enrollment:
-    | EnrollmentFieldsFragment
-    | HouseholdClientFieldsFragment['enrollment']
-    | ClientEnrollmentFieldsFragment
-    | ProjectEnrollmentFieldsFragment;
-  hideIcon?: boolean;
-  withActiveRange?: boolean;
-  activeColor?: Colors;
-  closedColor?: Colors;
-}) => {
-  let Icon = TimerIcon;
-
-  let text = 'Exited';
-  let textColor = closedColor;
+interface CommonStatusProps {
+  variant: 'inProgress' | 'open' | 'autoExited' | 'exited';
+}
 
-  if (enrollment.inProgress) {
-    Icon = ErrorOutlineIcon;
-    text = 'Incomplete';
-    textColor = 'error';
-  } else if (enrollment.autoExited) {
-    text = 'Auto-Exited';
-  } else if (!enrollment.exitDate) {
-    Icon = HistoryIcon;
-    text = 'Open';
-    textColor = activeColor;
-  }
+const CommonStatus: React.FC<CommonStatusProps> = ({ variant }) => {
+  const statusProps = useMemo(() => {
+    switch (variant) {
+      case 'inProgress':
+        return {
+          Icon: ErrorOutlineIcon,
+          text: 'Incomplete',
+          textColor: 'error',
+        };
+      case 'open':
+        return {
+          Icon: HistoryIcon,
+          text: 'Open',
+          textColor: 'activeStatus',
+        };
+      case 'autoExited':
+        return {
+          Icon: TimerIcon,
+          text: 'Auto-Exited',
+          textColor: 'text.secondary',
+        };
+      default:
+        return {
+          Icon: TimerIcon,
+          text: 'Exited',
+          textColor: 'text.secondary',
+        };
+    }
+  }, [variant]);
 
-  if (withActiveRange) {
-    const range = entryExitRange(
-      enrollment,
-      enrollment.inProgress ? 'Incomplete' : undefined
-    );
-    if (range) text = range;
-  }
   return (
     <Typography
       component='div'
       variant='body2'
-      color={textColor}
+      color={statusProps.textColor}
       sx={{ textDecoration: 'none' }}
     >
       <Stack direction='row' alignItems='center' gap={0.8}>
-        {!hideIcon && <Icon fontSize='small' />}
-        {text}
+        <statusProps.Icon fontSize='small' />
+        {statusProps.text}
       </Stack>
     </Typography>
   );
 };
 
+const EnrollmentStatus = ({
+  enrollment,
+}: {
+  enrollment:
+    | EnrollmentFieldsFragment
+    | HouseholdClientFieldsFragment['enrollment']
+    | ClientEnrollmentFieldsFragment
+    | ProjectEnrollmentFieldsFragment;
+}) => {
+  if (enrollment.inProgress) return <CommonStatus variant='inProgress' />;
+  if (enrollment.autoExited) return <CommonStatus variant='autoExited' />;
+  if (!enrollment.exitDate) return <CommonStatus variant='open' />;
+  return <CommonStatus variant='exited' />;
+};
+
+export const HouseholdStatus = ({
+  household,
+}: {
+  household: HouseholdWithEnrollmentFieldsFragment;
```

Comment:
Might be better to loosen these constraints a bit, since only 2 fields are needed. Then we can use this component in other places and not worry about type issues.

```suggestion
  household: Pick<Household, 'anyInProgress' | 'latestExitDate'>;
```
The same could be done for `EnrollmentStatus` component above as well. I think that approach would be more maintainable than jamming in all the fragments we happen to use.